File: app/src/main/java/com/esoxjem/movieguide/listing/sorting/SortingDialogInteractor.java
package com.esoxjem.movieguide.listing.sorting;

/**
 * @author arun
 */
public class SortingDialogInteractor implements ISortingDialogInteractor
{
    private SortingOptionStore sortingOptionStore;

    public SortingDialogInteractor(SortingOptionStore store)
    {
        sortingOptionStore = store;
    }

    @Override
    public int getSelectedSortingOption()
    {
        return sortingOptionStore.getSelectedOption();
    }

    @Override
    public void setSortingOption(SortType sortType)
    {
        sortingOptionStore.setSelectedOption(sortType);
    }
}

File: app/src/main/java/com/esoxjem/movieguide/movies/sorting/SortingDialogInteractor.kt
package com.esoxjem.movieguide.movies.sorting

/**
 * @author arunsasidharan
 */
internal class SortingDialogInteractor(
        private val sortingOptionStore: SortingOptionStore) :
        SortingContract.Interactor {

    override fun getSelectedSortingOption() = sortingOptionStore.getSelectedOption()

    override fun setSortingOption(sortType: SortType) = sortingOptionStore
            .setSelectedOption(sortType)
}
